sunpy/net/dataretriever/sources/stereo.py

```diff
@@ -0,0 +1,731 @@
+
+    @classmethod
+    def _can_handle_query(cls, *query, **kwargs):
+        """
+        Answers whether client can service the query.
+        Parameters
+        ----------
+        query : list of query objects
+        
+        Returns
+        -------
+        boolean
+            answer as to whether client can service the query
+            
+        """
+        chkattr =  ['Time', 'Instrument']
+        chklist =  [x.__class__.__name__ in chkattr for x in query]
+        for x in query:
+            if x.__class__.__name__ == 'Instrument' and x.value == 'stereo/het':
+                return all(chklist)
+        return False
+
+
+class SITClient(GenericClient):
+
+    def _get_url_for_timerange(cls,timerange, stereo_spacecraft = 'ahead', 
+                               species       = '4He', 
+                               duration_of_average = 10 * u.min):
+        """
+        Returns list of URLS to STEREO SIT data files corresponding to value of input timerange.
+        URL Source : http://www.srl.caltech.edu/STEREO/DATA/SIT/
+
+        The earliest data available is from Jan 2007.
+
+        Parameters
+        ----------
+        timerange: sunpy.time.TimeRange
+            time range for which data is to be downloaded.
+            Example value -  TimeRange('2007-01-01','2015-03-01')   
+
+        stereo_spacecraft: string   
+            Default value - ahead
+            Possible values - ahead, behind    # corresponding to spacecraft location
+
+        species:  string
+            Default value - 4He
+            Possible values - 4He, Fe, H, O
+
+        duration_of_average: string
+        Default value - 15min
+            Possible values - 1min, 10min, 1hr, 1day        #corresponding to duration over which data is averaged
+
+        Returns
+        -------
+        urls : list
+            list of URLs corresponding to the requested time range
+
+        Examples
+        --------
+        >>> from sunpy.time.timerange import TimeRange
+        >>> from sunpy.net.vso.attrs import Time, Instrument
+        >>> import sunpy.net.dataretriever.sources.stereo as stereo
+        >>> LCClient = stereo.SITClient()
+
+        >>> qr1 = LCClient.query(Time(TimeRange('2008/03/01', '2008/07/02')), Instrument('stereo/sit'), species = '4He', 
+                                    stereo_spacecraft = 'ahead', duration_of_average = 10 * u.min)
+        >>> res = LCClient.get(qr1)
+        >>> download_list = res.wait()
+
+        References
+        ----------
+        | http://www.srl.caltech.edu/STEREO/DATA/SIT/
+
+        """
+
+        possible_spacecraft = ['ahead', 'behind']
+        possible_duration   = [1 * u.min, 10 * u.min, 1 * u.h, 1 * u.d]
+        possible_species     = ['4He', 'Fe', 'H','O']
+
+        dict_duration     =   { 1 * u.min :'1min', 10 * u.min : '10min', 1 * u.h : '1hr', 1 * u.d :'1day'}
+        dict_time         =   { 1 * u.min :'%j' , 10 * u.min : '%m',1 * u.h : '',1 * u.d : '' }
```

Comment:
oh thats because the file pattern is like that : http://www.srl.caltech.edu/STEREO/DATA/SIT/ahead/1min/4He/ 

1 min files have %j pattern 10 min files %m so url pattern had to be generated this way.